### ISSUE TITLE
PQTest.exe run-test dies when encountering a syntax error

### ISSUE THREAD

**Issue by bgribaudo** [microsoft/vscode-powerquery-sdk#170]
### Preflight Checklist

- [X] I have installed the [latest version of Power Query SDK](https://github.com/microsoft/vscode-powerquery-sdk/releases/latest).
- [X] I have checked existing resources, including the [common issues](https://learn.microsoft.com/en-us/power-query/commonissues) and the [release notes](https://github.com/microsoft/vscode-powerquery-sdk/releases).
- [X] I have searched for [similar issues](https://github.com/microsoft/vscode-powerquery-sdk/issues).

### Power Query SDK

0.1.7

### Platform

Windows

### Architecture

x64

### Bug Description

If a test query's contents do not conform to M's synatx rules, `PQTest.exe run-test` dies (exits abnormally). 

While syntax errors are erroneous, unfortunately they are anticipated to occur during development, so should be gracefully handled.

### Steps to Reproduce

In the SDK, change the test query's contents to be `1 +`, then click `Evaluate current file`.

### Actual Experience

````
PQTest.exe run-test
  --extension "c:\Users\ben\Desktop\TestConnector\BenTest\bin\AnyCPU\Debug\BenTest.mez"
  --queryFile "c:\Users\ben\Desktop\TestConnector\BenTest\BenTest.query.pq"
  --prettyPrint
[11:14:00 AM]	[Error]	[Task exited abnormally] pqtest run-test pid(21500) exit(4294967291) stderr: Error: exception while running expression.: Token Literal expected. Start position: (2, 28). End position (2, 29).
````

### Expected Experience

* `PQTest.exe run-test` should complete, gracefully reporting details of the error using a similar JSON format to what it outputs when evaluation completes successfully.
* The syntax error should be reported in the `PQTest result` pane, similar to how an error raised by an M expression is displayed.

### Additional Context

Rough idea for how lexer/parser errors could be reported out by `pqTest`: 

````
[11:16:38 AM]	[Info]	RunTestBattery result [
    {
        "StartTime": "2022-10-19T15:16:37.3480863+00:00",
        "EndTime": "2022-10-19T15:16:38.3207362+00:00",
        "Method": "PQTest.RunTest",
        "Type": "PQTest.Expression"
        "IsSuccessfulEvaluation": false,
        "Status": "LexerParserError",
        "LexerParserError": {
           "Detail":  {same/similar structure goes here to what would be returned if the syntax error occurred inside a call to Expression.Evaluate -- e.g. Evaluate.Expression("1 +")}
           "Origin": {
              "Type": "File",
              "FilePath": "c:\Users\ben\Desktop\TestConnector\BenTest\BenTest.query.pq"
              "Location": { "StartPosition": "2, 28", "EndPosition": "2, 29" }
           }
        }
    }
]
````

If, instead, the lexer/parser error came from inside the packaged extension, then the `Origin` could change to something along the lines of:
````
           "Origin": {
              "Type": "Extension",
              "ExtensionName": "BenTest",
              "ExtensionPackage": "c:\Users\ben\Desktop\TestConnector\BenTest\bin\AnyCPU\Debug\BenTest.mez"
              "Location": { "StartPosition": "2, 28", "EndPosition": "2, 29" }
           }
````

**Comment by mattmasson** [comment]
This should be resolved in the [0.3.1 release](https://github.com/microsoft/vscode-powerquery-sdk/releases/tag/v0.3.1) (sdktools `2.127.2`).

```json
  {
      "ActivityId": "785b203d-6cf3-4ade-af76-f09e6d0d8422",
      "DataSourceAnalysis": [],
      "Details": "Token ',' expected. Start position: (3, 22). End position (3, 38).",
      "EndTime": "2024-03-18T22:05:19.6392845+00:00",
      "Method": "PQTest.RunTest",
      "Name": "test.query.pq",
      "StartTime": "2024-03-18T22:05:19.6228897+00:00",
      "DiagnosticEvents": {},
      "RowCount": 0,
      "Status": "Failed",
      "Type": "PQTest.Expression",
      "Error": {
          "Message": "Token ',' expected. Start position: (3, 22). End position (3, 38)."
      }
  }
```

**Comment by bgribaudo** [comment]
Thank you, @mattmasson and team!
